src/diffusers/schedulers/scheduling_ddim.py

```diff
@@ -315,14 +338,13 @@ def step(
             )
 
         # 4. Clip or threshold "predicted x_0"
-        if self.config.clip_sample:
+        if self.config.thresholding:
+            pred_original_sample = self._threshold_sample(pred_original_sample)
+        elif self.config.clip_sample:
             pred_original_sample = pred_original_sample.clamp(
                 -self.config.clip_sample_range, self.config.clip_sample_range
             )
```

Comment:
We want to do either regular -1, 1 sample clipping, or the threshold clipping which does the -1,1 clipping as a side case.

Note that it's important these are in one conditional as an if-else because the clip_sample config defaults to True and it could be easy for someone to only set the thresholding config to true and forget to set the clip_sample to False.

Reply:
make sense!